### PR TITLE
[prompty] expose prompty.render to get the prompt template

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
```python
prompty = Prompty.load(source="prompty_example.prompty")
result = prompty.render(question="what is the result of 1+1?")
```
Return value(str):
```
[{"role": "system", "content": "You are an AI assistant who helps people find information.\nAs the assistant, you answer questions briefly, succinctly,\nand in a personable manner using markdown and even add some personal flair with appropriate emojis.\n\n# Safety\n- You **should always** reference factual statements to search results based on [relevant documents]\n- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions\n# Customer\nYou are helping John Doh to find answers to their questions.\nUse their name to address them in your responses."}, {"role": "user", "content": "what is the result of 1+1?"}]
```

User can leverage this practice to calculate token used:
image

reference: 
- https://platform.openai.com/tokenizer
- https://github.com/openai/tiktoken

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/core/_flow.py [hunk 1]
```diff
@@ -417,6 +421,18 @@ def __call__(self, *args, **kwargs):
             outputs=self._outputs,
         )
 
+    def render(self, *args, **kwargs):
+        """Render the prompt template.
+
+        :param args: positional arguments are not supported.
+        :param kwargs: prompty inputs with key word arguments.
+        :return:
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/_flow.py:429]: document the return value

Comment (lalala123123) [reply src/promptflow-core/promptflow/core/_flow.py]: update docstring

File: src/promptflow-core/promptflow/core/_flow.py [hunk 2]
```diff
@@ -417,6 +421,19 @@ def __call__(self, *args, **kwargs):
             outputs=self._outputs,
         )
 
+    def render(self, *args, **kwargs):
+        """Render the prompt content.
+
+        :param args: positional arguments are not supported.
+        :param kwargs: prompty inputs with key word arguments.
+        :return: Prompt content
+        :rtype: Union[str, list]
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/_flow.py:430]: when will this return a str?

Comment (lalala123123) [reply src/promptflow-core/promptflow/core/_flow.py:430]: when the api type is completion, return vlaue is a string.

**Review by wangchao1230 — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**